src/main/java/org/dependencytrack/resources/v1/ProjectResource.java

```diff
@@ -309,6 +315,32 @@ public Response createProject(Project jsonProject) {
                 Project parent = qm.getObjectByUuid(Project.class, jsonProject.getParent().getUuid());
                     jsonProject.setParent(parent);
             }
+            final List<Team> choosenTeams = jsonProject.getAccessTeams();
+            Principal principal = getPrincipal();
+            List<Team> userTeams = new ArrayList<Team>();
+            if (principal instanceof final UserPrincipal userPrincipal) {
+                userTeams = userPrincipal.getTeams();
+            } else if (principal instanceof final ApiKey apiKey) {
+                userTeams = apiKey.getTeams();
+            }
+            boolean required = qm.isEnabled(ConfigPropertyConstants.ACCESS_MANAGEMENT_ACL_ENABLED);
+            boolean isAdmin = qm.hasAccessManagementPermission(principal);
+            if (required && choosenTeams.isEmpty()) {
+                return Response.status(422)
+                        .entity("You need to specify at least one team to which the project should belong").build();
+            }
+            List<Team> visibleTeams = isAdmin ? qm.getTeams() : userTeams;
+            List<UUID> visibleUuids = visibleTeams.isEmpty() ? new ArrayList<UUID>(): visibleTeams.stream().map(Team::getUuid).toList();
+            jsonProject.setAccessTeams(new ArrayList<Team>());
+            for (Team choosenTeam : choosenTeams) {
+                if (!visibleUuids.contains(choosenTeam.getUuid())) {
+                    return isAdmin ? Response.status(404).entity("This team does not exist!").build()
+                            : Response.status(403)
+                                    .entity("You don't have the permission to assign this team to a project.").build();
+                }
+                Team ormTeam = qm.getObjectByUuid(Team.class, choosenTeam.getUuid());
+                jsonProject.addAccessTeam(ormTeam);
+            }
```

Comment:
In the spirit of work avoidance, try to only execute this logic when really necessary. Also, keep in mind that existing clients will not submit `accessTeams`, so you have to expect it to be `null`.

If ACL is not enabled, this code doesn't need to run. Checking permissions and fetching available teams would be wasted work.

If ACL *is* enabled, but `jsonProject.getAccessTeams()` is `null` or empty, you can short-circuit immediately and return a `422` response (a `400` response would be more consistent with other API endpoints).

Reply:
> If ACL is not enabled, this code doesn't need to run. Checking permissions and fetching available teams would be wasted work.
I thought that it would be useful, that you can add teams even though its not enabled. So I would change it to only be run if accessTeams is not null/empty